src/renderers/WebGLRenderer.js

```diff
@@ -2440,8 +2440,8 @@ class WebGLRenderer {
 
 			}
 
-			const width = sourceBox.max.x - sourceBox.min.x + 1;
-			const height = sourceBox.max.y - sourceBox.min.y + 1;
+			const width = Math.round( sourceBox.max.x - sourceBox.min.x );
+			const height = Math.round( sourceBox.max.y - sourceBox.min.y );
```

Comment:
Changing the way `width` and `height` is calculated impacts all usage of `copyTextureToTexture3D`, not just the compressed ones. This PR breaks existing code that relies on specifying exact texels to `_gl.*texSubImage3D`.

If the idea is to handle non-integer inputs in `sourceBox`, then at least it should say `Math.round(sourceBox.max.x - sourceBox.min.x + 1)`. Also, `sourceBox.min` is given directly to `_gl.pixelStorei`, as well as `position` to `_gl.*texSubImage3D`, so it still doesn't handle non-integer inputs.

Having the `width` and `height` calculation work differently from `depth` doesn't look right either.

I'd argue for reverting these two lines, but I also do not have extensive experience with the three.js codebase, so I might have missed something. Any thoughts, @RenaudRohlinger @Mugen87 ?

Reply:
> This PR breaks existing code that relies on specifying exact texels to _gl.*texSubImage3D.

Do you mind sharing a live example that shows how this PR breaks the previous usage of `copyTextureToTexture3D()`? A debugging possibility would make it easier to reevaluate this change.

> Having the width and height calculation work differently from depth doesn't look right either.

That is a fair point. I'm not sure why the computation needs to differ. 

@RenaudRohlinger Was the a specific reason just to update the width and height but not the depth one?

Reply:
(I started to make a live example but as this is a pretty niche/low-level function it became very contrived and not at all helpful in explaining this issue, so I'll let my previous comment stand. Let me know if anything is still unclear).